xl-objects.go

```diff
@@ -190,9 +190,11 @@ func (xl xlObjects) GetObject(bucket, object string, startOffset int64) (io.Read
 			// Reset offset to 0 as it would be non-0 only for the first loop if startOffset is non-0.
 			offset = 0
 			if _, err := io.Copy(fileWriter, r); err != nil {
```

Comment:
err here has a shadowing issue with the top level  `r, err :=`  use it like 

```
if _, err = io.Copy(fileWriter, r)
```

Yes it is true that go tools do not complain about these all the time in a consistent manner.